akka-actor/src/main/scala/akka/actor/Address.scala

```diff
@@ -67,8 +66,7 @@ final case class Address private[akka] (protocol: String, system: String, host:
    */
   def hasGlobalScope: Boolean = host.isDefined
   // store hashCode
-  @nowarn("msg=deprecated")
-  @transient override lazy val hashCode: Int = scala.util.hashing.MurmurHash3.productHash(this)
+  @transient override lazy val hashCode: Int = scala.util.hashing.MurmurHash3.caseClassHash(this)
```

Comment:
Let's stay with the deprecated `productHash`. caseClassHash doesn't exist in earlier Scala library versions so if someone ends up running with earlier Scala version this will be a fatal error.